sdk/textanalytics/azure-ai-textanalytics/azure/ai/textanalytics/_text_analytics_client.py

```diff
@@ -489,5 +501,11 @@ def analyze_sentiment(  # type: ignore
                 cls=kwargs.pop("cls", sentiment_result),
                 **kwargs
             )
+        except TypeError as error:
+            if "opinion_mining" in str(error):
+                raise NotImplementedError(
+                    "'show_opinion_mining' is only available for API version v3.1-preview.1 and up"
```

Comment:
early, but we should create an issue to update this value once service GA`s 3.1

Reply:
issue here: https://github.com/Azure/azure-sdk-for-python/issues/12832